LifeSpace/Debug/LogManager.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  LogStore.swift
+//  LifeSpace
+//
+//  Created by Vishnu Ravi on 11/8/24.
+//
+
+import Foundation
+import OSLog
+import Spezi
+import SwiftUI
+
+actor LogManager: Module, DefaultInitializable, EnvironmentAccessible {
+    @Application(\.logger) private var logger
+    
+    func query(
+        startDate: Date? = nil,
+        endDate: Date? = nil,
+        logType: OSLogEntryLog.Level? = nil
+    ) -> String {
+        do {
+            let store = try OSLogStore(scope: .currentProcessIdentifier)
+            let position: OSLogPosition
+            if let startDate = startDate {
+                position = store.position(date: startDate)
+            } else {
+                position = store.position(timeIntervalSinceLatestBoot: 1)
+            }
+            
+            let logs = try store.getEntries(at: position).compactMap { $0 as? OSLogEntryLog }
```

Comment:
I think you might want to use the NSPredicates you can pass in here to do the filtering you do manually below.
Apple's documentation is very poor on this but there seem to be people who have figured most of these predicates out: https://useyourloaf.com/blog/fetching-oslog-messages-in-swift/

And you might want to get a reverse order to ensure that the latest entries are shown first: https://developer.apple.com/documentation/oslog/oslogenumerator/options/reverse